packages/next/src/server/render-result.ts

```diff
@@ -279,6 +290,18 @@ export default class RenderResult<
    * @param res
    */
   public async pipeToNodeResponse(res: ServerResponse) {
-    await pipeToNodeResponse(this.readable, res, this.waitUntil)
+    await pipeToNodeResponse(
+      this.readable,
+      res,
+      // Create a promise that will only run this callback when it's awaited. It
+      // will start the queue and wait until all the tasks are done.
```

Comment:
why not schedule it using `res.on('close', () => ...)`? maybe i'm missing something but `scheduleLazily` seems a bit brittle

Reply:
We want this to start _before_ the response closes, not after. In lambda environments where `waitUntil` is supported, this will utilize the fetch event's `waitUntil` implementation.